src/lio/bin/lio_rm.c

```diff
@@ -100,50 +109,56 @@ int main(int argc, char **argv)
         goto finished;
     }
 
-
     //** Spawn the tasks
-    n = argc - start_index;
-    tbx_type_malloc(flist, lio_path_tuple_t, n);
-    tbx_type_malloc(rpath, lio_os_regex_table_t *, n);
+    tbx_type_malloc(flist, lio_path_tuple_t, lio_parallel_task_count);
+    tbx_type_malloc(rpath, lio_os_regex_table_t *, lio_parallel_task_count);
+    tbx_type_malloc(path_list, char *, lio_parallel_task_count);
 
     q = gop_opque_new();
     opque_start_execution(q);
-    for (i=0; i<n; i++) {
-        flist[i] = lio_path_resolve(lio_gc->auto_translate, argv[i+start_index]);
+    i = 0;
+    loop = 0;
+    while ((path = lio_stdinlist_iter_next(piter)) != NULL) {
+        loop++;
```

Comment:
Please elaborate.  What's weird about it?  It's a loop over all the 
paths with a cap on the concurrency.